consensus/clique/src/test/java/tech/pegasys/pantheon/consensus/clique/CliqueVotingBlockInterfaceTest.java

```diff
@@ -0,0 +1,155 @@
+        .isEqualTo(Optional.empty());
+  }
+
+  @Test
+  public void headerWithNonceOfZeroReportsDropVote() {
+    final BlockHeaderTestFixture headerBuilder = new BlockHeaderTestFixture();
+    headerBuilder.coinbase(AddressHelpers.ofValue(1)).nonce(0L);
+    final CliqueVotingBlockInterface serDeser = new CliqueVotingBlockInterface();
+
+    final BlockHeader header =
+        TestHelpers.createCliqueSignedBlockHeader(headerBuilder, proposerKeys, validatorList);
+    final Optional<CastVote> extractedVote = serDeser.extractVoteFromHeader(header);
+
+    assertThat(extractedVote).isNotEqualTo(Optional.empty());
+    assertThat(extractedVote.get().getRecipient()).isEqualTo(header.getCoinbase());
+    assertThat(extractedVote.get().getVotePolarity()).isEqualTo(ValidatorVotePolarity.DROP);
+    assertThat(extractedVote.get().getProposer()).isEqualTo(proposerAddress);
+  }
+
+  @Test
+  public void headerWithNonceOfMaxLongReportsAddVote() {
+    final BlockHeaderTestFixture headerBuilder = new BlockHeaderTestFixture();
+    headerBuilder.coinbase(AddressHelpers.ofValue(2)).nonce(0xFFFFFFFFFFFFFFFFL);
+    final CliqueVotingBlockInterface serDeser = new CliqueVotingBlockInterface();
+
+    final BlockHeader header =
+        TestHelpers.createCliqueSignedBlockHeader(headerBuilder, proposerKeys, validatorList);
+    final Optional<CastVote> extractedVote = serDeser.extractVoteFromHeader(header);
+
+    assertThat(extractedVote).isNotEqualTo(Optional.empty());
+    assertThat(extractedVote.get().getRecipient()).isEqualTo(header.getCoinbase());
+    assertThat(extractedVote.get().getVotePolarity()).isEqualTo(ValidatorVotePolarity.ADD);
+    assertThat(extractedVote.get().getProposer()).isEqualTo(proposerAddress);
+  }
+
+  @Test
+  public void blendingAddVoteToHeaderResultsInHeaderWithNonceOfMaxLong() {
+    final BlockHeaderTestFixture headerBuilder = new BlockHeaderTestFixture();
+
+    final BlockHeaderBuilder builder =
+        BlockHeaderBuilder.fromHeader(headerBuilder.buildHeader())
+            .blockHashFunction(MainnetBlockHashFunction::createHash);
+
+    final CliqueVotingBlockInterface serDeser = new CliqueVotingBlockInterface();
+
+    CastVote vote =
+        new CastVote(
+            ValidatorVotePolarity.ADD, AddressHelpers.ofValue(1), AddressHelpers.ofValue(2));
+    final BlockHeaderBuilder builderWithVote =
+        serDeser.insertVoteToHeaderBuilder(builder, Optional.of(vote));
+
+    final BlockHeader header = builderWithVote.buildBlockHeader();
+
+    assertThat(header.getCoinbase()).isEqualTo(vote.getRecipient());
+    assertThat(header.getNonce()).isEqualTo(0xFFFFFFFFFFFFFFFFL);
+  }
+
+  @Test
+  public void blendingDropVoteToHeaderResultsInHeaderWithNonceOfZero() {
+    final BlockHeaderTestFixture headerBuilder = new BlockHeaderTestFixture();
+
+    final BlockHeaderBuilder builder =
+        BlockHeaderBuilder.fromHeader(headerBuilder.buildHeader())
+            .blockHashFunction(MainnetBlockHashFunction::createHash);
```

Comment:
This could probably be a class field as well to reduce noise in the actual test method.

Reply:
done.